src/Database/PostgreSQL/PQTypes/Checks.hs

```diff
@@ -830,16 +830,25 @@ checkDBConsistency options domains tablesWithVersions migrations = do
           runQuery_ (sqlDropIndexConcurrently tname idx) `finally` begin
           updateTableVersion
 
-        ModifyColumnMigration cursorSql updateSql batchSize -> do
+        ModifyColumnMigration tableName cursorSql updateSql batchSize -> do
           logMigration
           withCursorSQL "migration_cursor" NoScroll Hold cursorSql $ \cursor -> do
-            fix $ \loop -> do
-              cursorFetch_ cursor (CD_Forward batchSize)
-              primaryKeys <- fetchMany id
-              unless (null primaryKeys) $ do
-                updateSql primaryKeys
-                commit
-                loop
+            -- Vacuum should be done approximately once every 5% of the table had been
+            -- updated.
+            -- TODO: Explain more. Explain why.
```

Comment:
You want me to write this comment ? :)

Reply:
@marco44 Yes, please! I mean you can post it here and I can incorporate it with the next round of changes, which I'm going to do anyway.

Reply:
I think we should also specify a minimum amount of rows to consider for this periodic vacuuming, since I guess there is a minimum.

Reply:
Ok
In PostgreSQL, when a record is updated, a new version of this record is created. The old one is destroyed by the "vacuum" command when no transaction needs it anymore. So there's an autovacuum daemon whose purpose is to do this cleanup, and this is sufficient most of the time. It's tuned to try to keep the "bloat" (dead records) at around 10% of the table size in our environment. But it's also tuned to not saturate the server with IOs while doing the vacuum (it's IO intensive as there are a lot of reads and rewrites), as IOs on cloud server are really slow (and costly). So autovacuum won't be able to keep up with an aggressive batch update. So we need to run vacuum ourselves, to make sure things keep in check. The 5% limit is arbitrary, but a reasonable ballpark estimate: it more or less makes sure we keep dead records in the 10% envelope and the table doesn't grow too much during the operation.

Reply:
The mininum is "1". But yeah, of course we wouldn't vacuum for 1 record. The autovacuum "minimum" threshold is 50, I think this is a reasonable figure

Reply:
Thanks a ton! I slightly reworded it for the purposes of a public library.